apps/studio/src/components/Searchbar/SearchResult.tsx

```diff
@@ -0,0 +1,151 @@
+import { Box, HStack, Icon, Skeleton, Text, VStack } from "@chakra-ui/react"
+import { ResourceType } from "~prisma/generated/generatedEnums"
+
+import type { SearchResultResource } from "~/server/modules/resource/resource.types"
+import { ICON_MAPPINGS } from "~/features/dashboard/components/DirectorySidebar/constants"
+import { getLinkToResource } from "~/utils/resource"
+
+const formatDate = (date: Date): string => {
+  const now = new Date()
+  const diffInMs = now.getTime() - date.getTime()
+  const diffInDays = Math.floor(diffInMs / (1000 * 60 * 60 * 24))
+
+  if (diffInDays === 0) {
+    return "today"
+  } else if (diffInDays === 1) {
+    return "yesterday"
+  } else if (diffInDays >= 2 && diffInDays <= 6) {
+    return `${diffInDays} days ago`
+  } else if (diffInDays >= 7 && diffInDays <= 14) {
+    return "last week"
+  } else {
+    // Format date as "DD MMM YYYY" for anything beyond 14 days
+    const options: Intl.DateTimeFormatOptions = {
+      day: "2-digit",
+      month: "short",
+      year: "numeric",
+    }
+    return date
+      .toLocaleDateString("en-GB", options)
+      .replace(/(\d{2}) (\w{3}) (\d{4})/, "$1 $2 $3")
+  }
+}
+
+export interface SearchResultProps {
+  siteId: string
+  item: SearchResultResource
+  searchTerms?: string[]
+  isLoading?: boolean
+  shouldHideLastEditedText?: boolean
+}
+
+export const SearchResult = ({
+  siteId,
+  item,
+  searchTerms = [],
+  isLoading = false,
+  shouldHideLastEditedText = false,
+}: SearchResultProps) => {
+  const { id, title, type, fullPermalink, lastUpdatedAt } = item
+  const titleWithHighlightedText: ReactNode[] = title
+    .split(" ")
+    .map((titleWord) => {
+      let matchingSearchTerm: string | null = null
+      for (const searchTerm of searchTerms) {
+        if (titleWord.toLowerCase().startsWith(searchTerm.toLowerCase())) {
+          matchingSearchTerm = searchTerm
+          break
+        }
+      }
+      const highlightedText: string = titleWord.slice(
+        0,
+        matchingSearchTerm?.length ?? 0,
+      )
+      const nonHighlightedText: string = titleWord.slice(
+        matchingSearchTerm?.length ?? 0,
+      )
+      return (
+        <Box display="flex" whiteSpace="nowrap">
+          {!!highlightedText && (
+            <Text
+              textStyle="subhead-2"
+              textColor="base.content.default"
+              backgroundColor="interaction.main-subtle.default"
+              borderRadius="base"
+            >
+              {highlightedText}
+            </Text>
+          )}
+          {!!nonHighlightedText && (
+            <Text textStyle="subhead-2" textColor="base.content.default">
+              {nonHighlightedText}
+            </Text>
+          )}
+        </Box>
+      )
+    })
+
+  const shouldShowLastEditedText: boolean =
+    !shouldHideLastEditedText &&
+    (type === ResourceType.Page ||
+      type === ResourceType.CollectionLink ||
+      type === ResourceType.CollectionPage)
```

Comment:
nit: I think this might be a good time to centralise the definition of a page vs a folder? Otherwise we will overlook places like these if we introduce new resource types (e.g. IndexPage)

Reply:
yeah i think it's a bit tricky because this function here we **do not** want to display "last edited" for all page-ish resources

nevertheless, has extracted out as `isAllowedToHaveLastEditedText` to hold as much single source of truth as possible

Reply:
addressed in https://github.com/opengovsg/isomer/pull/882/commits/a6c2a3645ba408e22c14f3e273db0fa3ab8c67ca